server/auth/auth.go

```diff
@@ -1039,7 +1039,7 @@ func (auth *AuthManager) loadUserScore(user account.AccountID) (int32, error) {
 func (auth *AuthManager) handleConnect(conn comms.Link, msg *msgjson.Message) *msgjson.Error {
 	connect := new(msgjson.Connect)
 	err := json.Unmarshal(msg.Payload, &connect)
```

Comment:
Can you switch these to `msg.Unmarshal` while you're here? These were written before the method existed, I think.

Reply:
That'll also point them towards the new documentation.

Reply:
Done. https://github.com/decred/dcrdex/pull/1014/commits/bd69d11f2661ca1845a47602ac7c99f69813e540

Also added unit tests for these edge cases with for the `(*Message).{Unmarshal,Response}` methods

Further, I decided to explicitly reject null payloads with the orderbook and unsub_orderbook unauthenticated routes, as well as in the the limit, market, and cancel routes handled by the OrderRouter since the zero-values are not sensible in any of these cases and we might as well be defensive.